test/server/http/admin_test.cc

```diff
@@ -45,4 +48,27 @@ TEST_F(AdminFilterTest, Trailers) {
   filter_.decodeTrailers(request_headers_);
 }
 
+// Can only get code coverage of AdminImpl::handlerCpuProfiler stopProfiler with
+// a real profiler linked in (successful call to startProfiler). startProfiler
+// requies tcmalloc.
+#ifdef TCMALLOC
+
+TEST_F(AdminFilterTest, AdminProfiler) {
+  Buffer::OwnedImpl data;
+  admin_.runCallback("/cpuprofiler?enable=y", data);
+  EXPECT_TRUE(Profiler::Cpu::profilerEnabled());
+  admin_.runCallback("/cpuprofiler?enable=n", data);
+  EXPECT_FALSE(Profiler::Cpu::profilerEnabled());
+}
+
+#endif
+
+TEST_F(AdminFilterTest, AdminBadProfiler) {
+  Buffer::OwnedImpl data;
+  AdminImpl admin_bad_profile_path("/dev/null", "/var/log/envoy/envoy.prof",
```

Comment:
/var/log/envoy/envoy.prof might be a valid path in some environments. Maybe use `/some/unlikely/bad/path.prof` for now, we should deal with this better when cleaning up the temp file story. Another approach would be to mock file handling, but I think this is a bit of clutter and overhead to put this dependency injection into AdminImpl.